### PR TITLE
GAI Bug Fix

### PR DESCRIPTION
**Pull request by asimks**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 [hunk 1]
```diff
@@ -116,17 +116,23 @@ class SVTBase: AzSKRoot
...
-			$cachedPolicyContent = [ConfigurationHelper]::PolicyCacheContent | Where-Object { $_.Name -eq $controlsJsonFileName }
-			if ($cachedPolicyContent)
+			if ([ConfigurationHelper]::PolicyCacheContent.ContainsKey($controlsJsonFileName))
 			{
-				$this.SVTConfig = $cachedPolicyContent.Content
-				if ($this.SVTConfig)
+				[Policy]$policy = [ConfigurationHelper]::PolicyCacheContent[$controlsJsonFileName]
+				$this.SVTConfig = $policy.Content
+				#If policy is already in Final State simply return
+				if ($policy.State -eq [PolicyStatus]::Final)
```
Comment (VishalHaibatpure) [inline src/AzSK.Framework/Abstracts/SVTBase.ps1:124]: Generic word "Final" does not indicate the intention of state

File: src/AzSK.Framework/Models/Enums.ps1 [hunk 2]
```diff
@@ -176,3 +176,16 @@ enum AssignmentType
 	Eligible
 	Active
 }
+
+enum PolicyStatus
```
Comment (VishalHaibatpure) [inline src/AzSK.Framework/Models/Enums.ps1:180]: Can we rename it to PolicyCacheStatus ?

File: src/AzSK.Framework/Abstracts/SVTBase.ps1 (hunk 1 above)
Comment (asimks) [reply src/AzSK.Framework/Abstracts/SVTBase.ps1:124]: Vishal we used Final to indicate all the expandable attributes have been expanded for the given configuration.

File: src/AzSK/Framework/Framework.ps1 [hunk 3]
```diff
@@ -77,7 +78,7 @@ $FrameworkPath =  ((Get-Item $PSScriptRoot).Parent.Parent).FullName +"\AzSK.Fram
 . $FrameworkPath\Managers\PartialScanManager.ps1
 
 . $FrameworkPath\Helpers\LogAnalyticsHelper.ps1
-. $FrameworkPath\Helpers\RemoteReportHelper.ps1
+#. $FrameworkPath\Helpers\RemoteReportHelper.ps1
```
Comment (VishalHaibatpure) [inline src/AzSK/Framework/Framework.ps1:81]: We can remove commented code if it is working after moving at top level

Comment (asimks) [reply src/AzSK/Framework/Framework.ps1:81]: ok

**Review by TarunKrShukla — APPROVED** [review]

**Review by Aboli-msft — APPROVED** [review]

**Pull request merged**